wandb/sdk/launch/builder/kaniko_builder.py

```diff
@@ -382,6 +427,21 @@ def _create_kaniko_job(
                     ),
                 )
             ]
+        if isinstance(self.registry, AzureContainerRegistry):
```

Comment:
I think a lot of these env/registry specific resource creations could be better done in methods attached to those classes, or at least put in helper functions. Where possible, adding a comment explaining why they are necessary.